### PR TITLE
[benchmark] Run benchmark as NativeActivity application on Android

### LINKED ISSUE #15977: Migrate android-benchmark-arm-v8 to next buildsystem
The final remaining blocker to finish up the removal of android platform code from this repository (see https://github.com/mapbox/mapbox-gl-native/pull/15970) is the reintroduction of `make android-benchmark-arm-v8` into the next build system for metric purposes. 

@tmpsantos can this be prioritized, this is the only remaining blocker to finish up the code removal.

cc @alexshalamov

### PR DESCRIPTION
**Pull request by jmalanen**
Fixes #15977

### CODE REVIEW

File: platform/android/src/test/benchmark_runner.cpp [hunk 1]
```diff
@@ -1,10 +1,313 @@
...
+ALooper* looper = NULL;
+
+void runner() {
+    std::vector<std::string> arguments = {"mbgl-benchmark-runner",
+                                          "--benchmark_repetitions=3",
+                                          "--benchmark_format=json",
+                                          "--benchmark_out=/sdcard/benchmark/results/results.json"};
+    std::vector<char*> argv;
+    for (const auto& arg : arguments) {
+        argv.push_back((char*)arg.data());
```
[the remark this answers is not in the document]
Comment (jmalanen) [reply platform/android/src/test/benchmark_runner.cpp]: Done

File: platform/android/src/test/benchmark_runner.cpp [hunk 2]
```diff
@@ -1,10 +1,313 @@
...
+    app->activity->vm->AttachCurrentThread(&env, NULL);
+
+    std::string storagePath(app->activity->internalDataPath);
+    std::string zipFile = storagePath + "/data.zip";
+
+    if (chdir("/sdcard")) {
+        mbgl::Log::Error(mbgl::Event::General, "Failed to change the directory to /sdcard");
+        app->activity->vm->DetachCurrentThread();
+        mbgl::Log::Info(mbgl::Event::General, "Close the App!");
+        return;
```
[the remark this answers is not in the document]
Comment (jmalanen) [reply platform/android/src/test/benchmark_runner.cpp]: Changed the failure handling

File: platform/android/src/test/benchmark_runner.cpp [hunk 3]
```diff
@@ -1,10 +1,312 @@
...
+                                          "--benchmark_out=/sdcard/benchmark/results/results.json"};
+    std::vector<char*> argv;
+    for (const auto& arg : arguments) {
+        argv.push_back(const_cast<char*>(arg.data()));
+    }
+    argv.push_back(nullptr);
+
+    mbgl::Log::Info(mbgl::Event::General, "Start BenchmarkRunner");
+    int status = mbgl::runBenchmark(argv.size(), argv.data());
+    mbgl::Log::Info(mbgl::Event::General, "BenchmarkRunner finished with status: '%d'", status);
```
Comment (zmiao) [inline platform/android/src/test/benchmark_runner.cpp]: What does this status mean? should we use status result as input for `changeState(env, app, true);`

Comment (jmalanen) [reply platform/android/src/test/benchmark_runner.cpp]: The returned status is actually always 0, so we shouldn't use it as input for changeState().

Comment (zmiao) [reply platform/android/src/test/benchmark_runner.cpp]: I see, then probably we do not need this variable.

File: platform/android/src/test/benchmark_runner.cpp [hunk 4]
```diff
@@ -1,10 +1,312 @@
...
+    size_t length = static_cast<size_t>(env->GetArrayLength(stringJbytes.get()));
+    jbyte* pBytes = env->GetByteArrayElements(stringJbytes.get(), NULL);
+
+    std::string ret = std::string(reinterpret_cast<char*>(pBytes), length);
+    env->ReleaseByteArrayElements(stringJbytes.get(), pBytes, JNI_ABORT);
+
+    return ret;
+}
+
+void unZipFile(JNIEnv* env, const std::string& zipFilePath, const std::string& destinationPath) {
```
Comment (tmpsantos) [inline platform/android/src/test/benchmark_runner.cpp]: This unzip file is the same as the code on test runner, and probably will get duplicated again on the unit-test runner. I think what we need is a "android test runtime". Meaning a shared wrapper for render tests, unit tests and benchmark tests.

Comment (jmalanen) [reply platform/android/src/test/benchmark_runner.cpp]: I agree, I actually already have a patch for that but was planning to upload it as a separate PR.

File: next/platform/android/android.cmake [hunk 5]
```diff
@@ -313,19 +313,59 @@ target_link_libraries(
...
 
 target_link_libraries(
     mbgl-benchmark-runner
     PRIVATE
         Mapbox::Base::jni.hpp
-        mapbox-gl
+        android
+        log
         mbgl-compiler-options
+        -Wl,--whole-archive
```
[the remark this answers is not in the document]
Comment (jmalanen) [reply next/platform/android/android.cmake]: Yes

**Pull request merged**